server/src/internalClusterTest/java/org/opensearch/gateway/RecoveryFromGatewayIT.java

```diff
@@ -797,7 +798,9 @@ public void testBatchModeEnabledWithoutTimeout() throws Exception {
         );
         assertTrue(ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_BATCH_MODE.get(internalCluster().clusterService().getSettings()));
         assertEquals(1, gatewayAllocator.getNumberOfStartedShardBatches());
-        assertEquals(1, gatewayAllocator.getNumberOfStoreShardBatches());
+        // Replica shard would be marked ineligible since there are no data nodes.
+        // It would then be removed from any batch and batches would get deleted, so we would have 0 replica batches
+        assertEquals(0, gatewayAllocator.getNumberOfStoreShardBatches());
```

Comment:
After starting one datanode and cluster turning yellow, can we validate for 0 started and stored shards ? Could we also add asserts on the Reroute explain `clusterRerouteResponse` ?

Reply:
Done